### PR TITLE
Add support for patching manifests

### PR DESCRIPTION
**Pull request by mprahl**
This adds support for patching manifests that are wrapped in a policy.
This uses Kustomize internally, and thus the patches must be in the
format of Kustomize patches. An exception is that if the patches are
applied to only a single manifest, the apiVersion, kind, metadata.name,
and metadata.namespace fields are not necessary. They will be copied
over from the manifest before the patches are applied using Kustomize.

### CODE REVIEW

**Comment by mprahl** [comment]
@dhaiducek and @ChunxiAlexLuo, I had to refactor some functions for this change and those changes are separated by commits, so it's easier to review this PR commit by commit.

File: internal/utils.go [hunk 1]
```diff
@@ -61,18 +59,50 @@ func getPolicyTemplate(policyConf *policyConfig) (
 				continue
 			}
 
-			manifests = append(manifests, *manifestFile...)
+			manifestFiles = append(manifestFiles, *manifestFile...)
+		}
+
```
Comment (ChunxiAlexLuo) [inline internal/utils.go:64]: All rest logic in getManifests and getPolicyTemplate looks good to me. After these refactorizations are merged, I also need to update my pr accordingly.

Thanks for so detailed comments!

File: docs/policygenerator-reference.yaml [hunk 2]
```diff
@@ -56,6 +56,14 @@ policies:
       # Required. Path to a single file or a flat directory of files relative to the
       # kustomization.yaml file.
       - path: ""
+        # Optional. A Kustomize patch to apply to the manifest(s) at the path. If there
+        # are multiple manifests, the patch requires the apiVersion, kind, metadata.name,
+        # and metadata.namespace (if applicable) fields to be set so Kustomize
+        # can identify the manifest the patch applies to.
+        patches:
```
[the remark this answers is not in the document]
Comment (ChunxiAlexLuo) [reply docs/policygenerator-reference.yaml:63]: Or set up an example that these fields required for multiple manifests case, like the logic in Validate()

[the remark this answers is not in the document]
Comment (ChunxiAlexLuo) [reply docs/policygenerator-reference.yaml:63]: Saw the update in policyGenerator.yaml

**Comment by mprahl** [comment]
@ChunxiAlexLuo thanks for the review! Could you please take another look? Here is the diff:
https://github.com/open-cluster-management/policy-generator-plugin/compare/9de0e28ea3ae7b38a7ed7ef0e1c0352d9e428fc6..e3386e178584bf3ac8a0ef32f1ad36edd73bc7d7

**Review by ChunxiAlexLuo — APPROVED** [review]
/lgtm

File: internal/plugin.go [hunk 3]
```diff
@@ -139,11 +140,21 @@ func (p *Plugin) Generate() ([]byte, error) {
...
+	i := 0
+	for k := range plrNameToPolicyIdxs {
+		plrNames[i] = k
+		i++
+	}
+	sort.Strings(plrNames)
+
 	plcBindingCount := 0
-	for plrName, policyIdxs := range plrNameToPolicyIdxs {
+	for _, plrName := range plrNames {
```
Comment (ChunxiAlexLuo) [inline internal/plugin.go:154]: /lgtm

**Review by ChunxiAlexLuo — APPROVED** [review]

**Pull request merged**